pkg/k8sutils/job.go

```diff
@@ -75,7 +76,7 @@ func (k8s *k8sImpl) CreateK8sJob(
 	automountServiceAccountToken := jobSettings.EnableKubernetesAPIAccess
 
 	// specify empty service account name for job
-	serviceAccountName := ""
+	serviceAccountName := jobSettings.DefaultJobServiceAccount
```

Comment:
Couple of things here:
- if we keep line 79 we should probably delete the comment in line 78 :wink: 
- Aren't we supposed to be backward-compatible?  If  we have `jobConfig.enableKubernetesApiAccess=true` we should specify the default service account

Reply:
This change doesn't affect the code path that will be taken if `jobConfig.enableKubernetesApiAccess=true`:
https://github.com/keptn-contrib/job-executor-service/blob/feat/restrict-job-service-account/pkg/k8sutils/job.go#L77-L83.

The service account will be overwritten as it is the case currently in the master branch. But a helm upgrade with `--reuse-values` might break backwards compatibility, because the new service account will be used.